pysph/base/z_order_gpu_nnps.pyx

```diff
@@ -119,22 +132,25 @@ cdef class ZOrderGPUNNPS(GPUNNPS):
 
         map_cid_to_idx = self.helper.get_kernel("map_cid_to_idx")
 
-        make_vec = cl.array.vec.make_double3 if self.use_double \
-                else cl.array.vec.make_float3
-
-        map_cid_to_idx(
+        event = map_cid_to_idx(
             pa_gpu.x, pa_gpu.y, pa_gpu.z,
             pa_wrapper.get_number_of_particles(), self.cell_size,
-            make_vec(self.xmin.data[0], self.xmin.data[1], self.xmin.data[2]),
+            self.make_vec(self.xmin[0], self.xmin[1], self.xmin[2]),
             self.pids[pa_index].array, self.pid_keys[pa_index].array,
             self.cids[pa_index].array, self.cid_to_idx[pa_index].array
         )
 
+        if self._profile:
+            profile("map_cid_to_idx", event)
+
         fill_cids = self.helper.get_kernel("fill_cids")
 
-        fill_cids(self.pid_keys[pa_index].array, self.cids[pa_index].array,
+        event = fill_cids(self.pid_keys[pa_index].array, self.cids[pa_index].array,
```

Comment:
A lot of this will be much nicer with a profile -- in fact, you can do this in `helper.get_kernel` directly.